src/testUtil/TestCase.ts

```diff
@@ -160,6 +167,11 @@ export class TestCase {
   }
 
   async recordFinalState(returnValue: unknown) {
+    if (returnValue instanceof Error) {
```

Comment:
I'd just record this one in `commandErrorHook`, as mentioned above.  Then you don't need to abuse `returnValue`